### PR TITLE
do not print empty values in docker info

### PR DESCRIPTION
**Pull request by vieux**
When jobs where removed in https://github.com/docker/docker/pull/12266 some 

`if value != ""` where omitted. It makes the docker info agains a swarm daemon looks bad.
This is my attempt to fix this.

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -286,3 +286,10 @@ func ImageReference(repo, ref string) string {
 func DigestReference(ref string) bool {
 	return strings.Contains(ref, ":")
 }
+
+// PrintfIfNotEmpty prints the string value if it's not empty
+func PrintfIfNotEmpty(w io.Writer, format, value string) {
```
Comment (runcom) [inline utils/utils.go]: Can we have this functions in `pkg/stringutils`?

Comment (tiborvass) [inline utils/utils.go]: I would put this in ioutils

Comment (runcom) [inline utils/utils.go]: oh yes, better

Comment (LK4D4) [inline utils/utils.go]: And this is `Fprintf`, not `Printf`.

**Comment by runcom** [comment]
I also think this needs a cli test to be sure of not regress anymore

**Comment by calavera** [comment]
agree with @runcom

**Comment by vieux** [comment]
@runcom @calavera I don't see how I can do the integration test, as it would require the engine to send a truncated API response. Any advice ?

**Comment by duglin** [comment]
You may be able to get away with just checking the output for any lines with a colon (:) but nothing else after it, except spaces - and if present then it failed.  And you could also start up a daemon w/o an HTTP proxy defined, that'll force at least one of your checks to kick in.

**Comment by duglin** [comment]
Or dummy an http server with the daemon output you want the CLI to see.

File: pkg/ioutils/fmt.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package ioutils
+
+import (
+	"fmt"
+	"io"
+)
+
+// FprintfIfNotEmpty prints the string value if it's not empty
+func FprintfIfNotEmpty(w io.Writer, format, value string) {
```
Comment (tiborvass) [inline pkg/ioutils/fmt.go]: should return `(int, error)`

Comment (vieux) [inline pkg/ioutils/fmt.go]: @tiborvass done

**Comment by tiborvass** [comment]
LGTM

**Comment by calavera** [comment]
LGTM, but janky is very sad :crying_cat_face:

**Comment by vieux** [comment]
@calavera not sad anymore :)

**Comment by calavera** [comment]
:tada:

**Comment by jfrazelle** [comment]
cherry-picked

**Pull request merged**